_includes/charts/tigera-operator/templates/tigera-operator/02-tigera-operator.yaml

```diff
@@ -42,3 +42,43 @@ spec:
               value: "tigera-operator"
             - name: TIGERA_OPERATOR_INIT_IMAGE_VERSION
               value: {{.Values.tigeraOperator.version}}
+{{- if eq .Values.kubernetesProvider "openshift" }}
+      initContainers:
+        - name: create-initial-resources
+          image: {{ join ":" (list .Values.calicoctl.image .Values.calicoctl.tag) }}
+          env:
+            - name: DATASTORE_TYPE
+              value: kubernetes
+          command:
+            - /bin/sh
+            - -c
+            - |-
+              if [ ! -d /calicoctl-resources ]; then
+                echo No calicoctl resources to install
+                exit 0
+              fi
+              for i in $(ls /calicoctl-resources/*); do
+                output=$(calicoctl create -f $i 2>&1)
+                if [ $? -eq 0 ]; then
+                  echo Created resource from $i
+                else
+                  echo $output | grep "resource already exists" > /dev/null
+                  if [ $? -eq 0 ]; then
+                    echo Resource already exists $i
+                  else
+                    echo Failed to create from $i, $output
+                    exit 1
+                  fi
+                fi
+              done
+              echo All calicoctl resources were created or exist
+          volumeMounts:
+            - name: calicoctl-resources
+              mountPath: /calicoctl-resources
+      volumes:
+        - name: calicoctl-resources
+          configMap:
+            defaultMode: 0400
+            name: calicoctl-resources
+            optional: true
```

Comment:
I'm not sure if this should be optional. Should we always have the ConfigMap defined and if the user does not add anything then this init container does nothing?

Reply:
I think we should make this mandatory - I'm worried that with optional, it introduces a race condition between the configmap existing and this script running where we might not apply the resources.